core/includes/form.inc

```diff
@@ -3729,6 +3702,63 @@ function theme_details($variables) {
   return $output;
 }
 
+/**
+ * Returns HTML for a fieldset form element and its children.
+ *
+ * @param $variables
+ *   An associative array containing:
+ *   - element: An associative array containing the properties of the element.
+ *     Properties used: #attributes, #children, #collapsed, #collapsible,
+ *     #description, #id, #title, #value.
+ *
+ * @ingroup themeable
+ */
+function theme_fieldset($variables) {
+  $element = $variables['element'];
+  element_set_attributes($element, array('id'));
+  _form_set_class($element, array('form-wrapper'));
+
+  $output = '<fieldset' . backdrop_attributes($element['#attributes']) . '>';
+  if (!empty($element['#title'])) {
+    // Always wrap fieldset legends in a SPAN for CSS positioning.
+    $output .= '<legend><span class="fieldset-legend">' . $element['#title'];
+    // If the fieldset contains any required fields in it, add a "required"
+    // indicator in the legend.
+    $output .= _element_has_required_children($element) ? ' <abbr class="form-required" title="' . t('This fieldset contains required fields.') . '">*</abbr>' : '';
+    $output .= '</span></legend>';
```

Comment:
This bit here is the only actual change in the `theme_fieldset()` function - it was otherwise only moved to come after the `theme_details()` function.